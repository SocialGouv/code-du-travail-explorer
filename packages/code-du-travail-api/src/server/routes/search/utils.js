// merge by one of each

const merge = (res1, res2, max_result) => {
  //  simple array concat
  // return [...res1.hits, ...res2.hits].sort(sortByScore);

  // interleave results
  const res = [];
  Array.from({
    length: max_result
  }).forEach((x, i) => {
    if (res1.length > i) res.push(res1[i]);
    if (res2.length > i) res.push(res2[i]);
  });
  return res;
};

// Remove Duplicates

const removeDuplicate = arr =>
  Object.values(
    arr.reduce(
      (values, current) => ({
        ...values,
        [current._source.source + "/" + current._source.slug]: current
      }),
      {}
    )
  );

const mergeDuplicate = arr => {
  return Object.values(
    arr.reduce((acc, current) => {
      const unique = current._source.source + "/" + current._source.slug;
      // case we've already seen the same result from a different algo
      if (
        acc[unique] != undefined &&
        acc[unique]._source.algo != current._source.algo
      ) {
        acc[unique]._source.algo = "both";
      } else {
        acc[unique] = current;
      }
      return acc;
    }, {})
  );
};

const mergePipe = (a, b, max_result) => {
  const res = merge(a, b, a.length + b.length);
  return mergeDuplicate(res).slice(0, max_result);
};

exports.merge = merge;
exports.removeDuplicate = removeDuplicate;
exports.mergePipe = mergePipe;
