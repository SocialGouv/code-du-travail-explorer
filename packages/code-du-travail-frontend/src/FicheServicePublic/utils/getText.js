export default function getText(element = { text: "" }, joint = " ") {
  if (element.type === "text") {
    return element.$.trim();
  }
  if (element.$) {
    return element.$.map(child => getText(child, joint)).join(joint);
  }
  return "";
}
