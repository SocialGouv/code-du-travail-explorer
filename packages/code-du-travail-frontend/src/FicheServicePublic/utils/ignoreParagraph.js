const ignoreParagraph = children =>
  children.map(child => {
    if (child.name === "Paragraphe") {
      return child.$;
    }
    return child;
  });

export default ignoreParagraph;
