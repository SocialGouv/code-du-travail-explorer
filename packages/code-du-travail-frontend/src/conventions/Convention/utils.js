import { matopush } from "../../piwik";

export function trackAccordionPanelState(conventionName, eventName) {
  let previousPanelState = [];
  return function (currentPanelState) {
    if (currentPanelState.length > previousPanelState.length) {
      const lastOpenedPanel = currentPanelState.filter(
        (panel) => !previousPanelState.includes(panel)
      );
      matopush(["trackEvent", eventName, conventionName, lastOpenedPanel[0]]);
    }
    previousPanelState = currentPanelState;
  };
}
