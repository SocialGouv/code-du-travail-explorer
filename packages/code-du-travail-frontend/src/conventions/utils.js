export const sortByIntOrdre = (a, b) => {
  if (a.intOrdre < b.intOrdre) {
    return -1;
  }
  if (a.intOrdre > b.intOrdre) {
    return 1;
  }
  return 0;
};
