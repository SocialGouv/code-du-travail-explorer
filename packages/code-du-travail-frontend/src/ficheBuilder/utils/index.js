export { default as ignoreParagraph } from "./ignoreParagraph";
export { default as getText } from "./getText";
