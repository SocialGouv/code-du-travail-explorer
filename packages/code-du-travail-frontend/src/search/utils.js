import { getRouteBySource, SOURCES } from "@cdt/sources";

import { matopush } from "../piwik";

export const summarize = (text = "") =>
  text.length > 160 ? text.slice(0, text.indexOf(" ", 160)) + "â€¦" : text;

export const formatUrlMatomo = (source, slug, url) => {
  const trackedUrl =
    source === SOURCES.EXTERNALS ? url : `/${getRouteBySource(source)}/${slug}`;

  return trackedUrl;
};

export const reportSelectionToMatomo = (source, slug, url, algo) => {
  const qualifiedCall = JSON.stringify({
    algo,
    url: formatUrlMatomo(source, slug, url),
  });

  matopush(["trackEvent", "selectResult", qualifiedCall]);
};
