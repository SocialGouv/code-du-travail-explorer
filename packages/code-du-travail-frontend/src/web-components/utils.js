export const displayInViewport = (target, rootBoundaries) => {
  const viewportHeight =
    window.innerHeight || document.documentElement.clientHeight;
  const viewportWidth =
    window.innerWidth || document.documentElement.clientWidth;

  target.style.display = "block";
  target.style.top = rootBoundaries.height + "px";

  let targetBounding = target.getBoundingClientRect();

  target.style.left =
    -Math.round((targetBounding.width - rootBoundaries.width) / 2) + "px";

  targetBounding = target.getBoundingClientRect();
  // we are overflowing on the right
  if (targetBounding.right > viewportWidth) {
    target.style.right = "0px";
    target.style.left = "auto";
  }
  // we are overflowing on the left
  if (targetBounding.left < 0) {
    target.style.left = "0px";
    target.style.right = "auto";
  }
  // we are overflwing to the bottom
  if (targetBounding.bottom > viewportHeight) {
    target.style.top = -Math.round(targetBounding.height) + "px"; // don't make it go below header
  }
  // second pass to make sure the first pass did not
  // make the tooltip overflow on the opposite side :o
  targetBounding = target.getBoundingClientRect();
  if (targetBounding.left < 0) {
    target.style.right = targetBounding.left - 10 + "px";
  }
  if (targetBounding.right > viewportWidth) {
    target.style.left = viewportWidth - targetBounding.right - 10 + "px";
  }
  // don't make it go below header
  if (target.getBoundingClientRect().top < 120) {
    target.style.top = rootBoundaries.height + "px";
  }
};

const throttler = function (fn, interval) {
  let ready = true;
  return function (...args) {
    if (!ready) return;
    if (ready) {
      ready = false;
      window.setTimeout(() => {
        ready = true;
      }, interval);
      return fn(...args);
    }
  };
};

export const throttledDisplayInViewport = throttler(displayInViewport, 10);
