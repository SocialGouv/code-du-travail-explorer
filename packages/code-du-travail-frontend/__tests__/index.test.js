import React from "react";
import { render } from "../test/utils";
import Home from "../pages/index.js";

describe("<Home />", () => {
  it("should render", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
