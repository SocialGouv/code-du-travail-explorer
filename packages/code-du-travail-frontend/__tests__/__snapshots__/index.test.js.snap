// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
.c19:not(:focus):not(:active) {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  border: 0;
}

.c42 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c35 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
  padding: 0;
  max-width: 46.25rem;
}

.c35 > *:first-child {
  margin-top: 0;
}

.c35 > *:last-child {
  margin-bottom: 0;
}

.c9 {
  padding: 1.25rem 0;
  background-color: #fff;
}

.c33 {
  padding: 1.25rem 0;
}

.c37 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #005994;
  border-color: #005994;
  border-bottom-color: #003a61;
}

.c37:not([disabled]):hover,
.c37:not([disabled]):focus {
  background: #0068ae;
  color: #fff;
}

.c37:not([disabled]):active {
  color: #fff;
  background: #0078c7;
  border-width: 2px 1px 1px 1px;
  border-color: #005994;
  outline: none;
}

.c37[aria-pressed="true"] {
  color: #fff;
  background: #0068ae;
  border-width: 2px 1px 1px 1px;
  border-color: #005994;
  border-top-color: #003a61;
  box-shadow: inset 0 1px 2px 0 #003a61;
}

.c37[aria-pressed="true"]:not([disabled]):hover,
.c37[aria-pressed="true"]:not([disabled]):focus {
  background-color: #005994;
}

.c37[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #003a61;
}

.c37[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

.c27 {
  padding: 0.625rem 1rem;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 100%;
}

.c30 {
  color: #4c5467;
  font-size: 0.875rem;
}

.c29 {
  color: #005994;
  font-size: 1.125rem;
}

.c28 {
  width: 2.5rem;
  display: inline-block;
  margin: 1rem;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  list-style-type: none;
  padding: 0;
  margin: NaN;
}

.c25 {
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: calc(100% / 4 - 2 * 0.625rem);
  -ms-flex-preferred-size: calc(100% / 4 - 2 * 0.625rem);
  flex-basis: calc(100% / 4 - 2 * 0.625rem);
  margin: 0.625rem;
  text-align: center;
}

.c7 {
  display: inline-block;
  padding: 0.5rem 0.625rem;
  font-size: 0.875rem;
  font-weight: bold;
  text-align: center;
  white-space: nowrap;
  line-height: 1rem;
  border-radius: 0.25rem;
  color: #0c0c0e;
  background: #e5f1fe;
}

.c18 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c21 ul {
  z-index: 100;
}

.c21 li[role="option"]:nth-child(2n + 1) {
  background: #f9f9fc;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin: 0 auto;
  padding: 0;
  overflow: visible;
}

.c17 {
  position: absolute;
  left: 0;
  top: 0;
  margin: 1rem 0 0 1.25rem;
  width: 1.3rem;
  height: 1.3rem;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0 1rem 0 4rem;
  width: 100%;
  height: 50px;
  font-size: inherit;
  font-family: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #f9f9fc;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  -webkit-transition: border 250ms ease;
  transition: border 250ms ease;
}

.c20:focus {
  border-color: #006be6;
  outline: none;
}

.c22 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #006be6;
  border-color: #006be6;
  border-bottom-color: #0053b3;
  margin-left: 0.5rem;
}

.c22:not([disabled]):hover,
.c22:not([disabled]):focus {
  background: #07f;
  color: #fff;
}

.c22:not([disabled]):active {
  color: #fff;
  background: #1a85ff;
  border-width: 2px 1px 1px 1px;
  border-color: #006be6;
  outline: none;
}

.c22[aria-pressed="true"] {
  color: #fff;
  background: #07f;
  border-width: 2px 1px 1px 1px;
  border-color: #006be6;
  border-top-color: #0053b3;
  box-shadow: inset 0 1px 2px 0 #0053b3;
}

.c22[aria-pressed="true"]:not([disabled]):hover,
.c22[aria-pressed="true"]:not([disabled]):focus {
  background-color: #006be6;
}

.c22[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #0053b3;
}

.c22[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

.c10 {
  padding-bottom: 2.5rem;
  background-color: #e5f1fe;
}

.c11 {
  margin-top: 0;
  line-height: 1.4;
  color: #005994;
}

.c12 {
  line-height: 1.1;
  margin-bottom: 0;
}

.c13 {
  font-weight: bold;
}

.c14 {
  display: block;
  margin-bottom: 0.625rem;
  font-size: 1.375rem;
  color: #005994;
}

.c15 {
  margin: 0;
  width: 750px;
}

.c0 {
  background: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.25rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c4:link {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  width: 5rem;
  height: 100%;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.125rem;
  line-height: 1.4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 {
  margin-left: 1.25rem;
}

.c34 {
  text-align: center;
}

.c36 {
  margin-bottom: 1.25rem;
  text-align: left;
}

.c38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c40 + .c39 {
  padding-left: 1rem;
}

.c41 {
  font-size: 1rem;
  font-weight: bold;
}

.c43 {
  margin: 0.5rem 0;
}

.c43 a,
.c43 a:visited {
  color: #434956;
}

.c31 {
  padding: 0;
  width: 100%;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: transparent;
  border: none;
  border-radius: 0.25rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
}

.c31 > * {
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c31:focus > *,
.c31:active > *,
.c31:hover > * {
  border: 1px solid #80bdff;
  box-shadow: 0 0 2px 2px #e5f1fe;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c23 {
  text-align: center;
  margin-bottom: 2rem;
}

.c26 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  height: 100%;
  border-radius: 0.25rem;
}

.c26 > * {
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c26:focus > *,
.c26:active > *,
.c26:hover > * {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  border: 1px solid #80bdff;
  box-shadow: 0 0 2px 2px #e5f1fe;
}

@media print {
  .c1 {
    max-width: 100%;
    padding: 0;
  }
}

@media print {
  .c35 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:980px) {
  .c25 {
    -webkit-flex-basis: calc(100% / 3 - 2 * 0.625rem);
    -ms-flex-preferred-size: calc(100% / 3 - 2 * 0.625rem);
    flex-basis: calc(100% / 3 - 2 * 0.625rem);
  }
}

@media (max-width:600px) {
  .c25 {
    -webkit-flex-basis: calc(100% - 2 * 0.625rem);
    -ms-flex-preferred-size: calc(100% - 2 * 0.625rem);
    flex-basis: calc(100% - 2 * 0.625rem);
  }
}

@media (max-width:600px) {
  .c16 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: auto;
  }
}

@media (max-width:600px) {
  .c17 {
    display: none;
  }
}

@media (max-width:600px) {
  .c20 {
    padding: 0 1rem;
  }
}

@media (max-width:600px) {
  .c22 {
    margin-left: 0;
    margin-top: 0.625rem;
  }
}

@media print {
  .c10 {
    display: none;
  }
}

@media (max-width:980px) {
  .c15 {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c15 {
    width: 100%;
    max-width: 400px;
  }
}

@media (max-width:980px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c2 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media print {
  .c32 {
    display: none;
  }
}

@media (max-width:600px) {
  .c38 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c40 + .c39 {
    padding-left: 0;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <a
          class="c4"
          href="/"
          title="Code du travail numérique - retour à l'accueil"
        >
          <img
            alt="symbole de la Marianne, site officiel du gouvernement"
            class="c5"
            src="/static/assets/img/marianne.svg"
          />
          <span
            class="c6"
          >
            Code du travail
            <br />
            numérique
          </span>
        </a>
        <div
          class="c7 c8"
        >
          Version Bêta
        </div>
      </div>
    </div>
  </header>
  <main
    id="main"
  >
    <div
      class="c9 c10"
    >
      <div
        class="c1"
      >
        <header
          class="c11"
        >
          <h1
            class="c12"
          >
            Bienvenue sur 
            <br />
            <span
              class="c13"
            >
              le Code du travail numérique
            </span>
          </h1>
          <p
            class=""
          >
            Vous cherchez une information sur le droit du travail ? Vous avez besoin d’accompagnement ?
            <br />
            Nous vous proposons des réponses accessibles et personnalisées selon votre situation.
          </p>
        </header>
        <label
          class="c14"
          for="searchbar"
        >
           Recherchez par mots clefs
        </label>
        <div
          class="c15"
        >
          <form
            class="c16"
          >
            <svg
              class="c17"
              viewBox="0 0 20 20"
            >
              <title />
              <path
                d="M17.876 8.939c0 2.197-.819 4.208-2.123 5.773l4.023 4.022a.72.72 0 0 1 0 1.043.735.735 0 0 1-.522.223.736.736 0 0 1-.52-.223l-4.023-4.023a8.813 8.813 0 0 1-5.772 2.123C4.022 17.877 0 13.855 0 8.939 0 4.022 4.023 0 8.939 0c4.915 0 8.937 4.023 8.937 8.939zM9.001 16.5c4.124 0 7.499-3.375 7.499-7.5S13.125 1.5 9 1.5C4.877 1.5 1.5 4.875 1.5 9s3.376 7.5 7.5 7.5z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            <label
              class="c18"
              for="searchbar"
            >
              <span
                class="c19"
              >
                Rechercher
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="react-autowhatever-cdtn-documents-suggester"
                role="combobox"
                style="flex: 1;"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-autowhatever-cdtn-documents-suggester"
                  autocomplete="off"
                  class="c20"
                  id="searchbar"
                  name="query"
                  placeholder="Recherche"
                  type="search"
                  value=""
                />
                <div
                  class="c21"
                  id="react-autowhatever-cdtn-documents-suggester"
                  role="listbox"
                />
              </div>
            </label>
            <button
              aria-pressed="false"
              class="c22"
              type="submit"
            >
              Rechercher
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <div
        class="c1"
      >
        <h2
          class="c23"
        >
          Découvrez nos outils
        </h2>
        <ul
          class="c24"
        >
          <li
            class="c25"
          >
            <a
              class="c26"
              href="/outils/indemnite-licenciement"
              title="Démarrer une simulation d'indemnités de licenciements"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/point-of-service_web.svg"
                />
                <h3
                  class="c29"
                >
                  Simulateur d'indemnités de licenciements
                </h3>
                <p
                  class="c30"
                >
                  Simulez simplement le montant d'une indemnité de licenciement en fonction de votre situation
                </p>
              </div>
            </a>
          </li>
          <li
            class="c25"
          >
            <a
              class="c26"
              href="/outils/preavis-demission"
              title="Démarrer une simulation de durée de préavis de démission"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/outils.svg"
                />
                <h3
                  class="c29"
                >
                  Simulateur de délais de préavis de démission
                </h3>
                <p
                  class="c30"
                >
                  Connaître la durée du préavis de démission.
                </p>
              </div>
            </a>
          </li>
          <li
            class="c25"
          >
            <a
              class="c26"
              href="/modeles-de-courriers"
              title="Voir tous les modèles de courriers"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/message_web.svg"
                />
                <h3
                  class="c29"
                >
                  Modèles de courriers
                </h3>
                <p
                  class="c30"
                >
                  Utilisez des modèles pré-remplis pour vos courriers liés au droit du travail
                </p>
              </div>
            </a>
          </li>
          <li
            class="c25"
          >
            <a
              class="c26"
              href="/outils/simulateur-embauche"
              title="Voir tous les modèles de courriers"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/salary_web.svg"
                />
                <h3
                  class="c29"
                >
                  Simulateur d'embauche
                </h3>
                <p
                  class="c30"
                >
                  Estimez le salaire lors d'une embauche : total employeur, brut, net
                </p>
              </div>
            </a>
          </li>
          <li
            class="c25"
          >
            <a
              class="c26"
              href="/outils/indemnite-precarite"
              title="Calculez la prime de précarité"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/coins.svg"
                />
                <h3
                  class="c29"
                >
                  Prime de précarité
                </h3>
                <p
                  class="c30"
                >
                  Simulez la prime de précarité d’un salarié.
                </p>
              </div>
            </a>
          </li>
          <li
            class="c25"
          >
            <button
              class="c31"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/book_web.svg"
                />
                <h3
                  class="c29"
                >
                  Votre convention collective
                </h3>
                <p
                  class="c30"
                >
                  Trouvez simplement la convention collective dont vous dépendez
                </p>
              </div>
            </button>
          </li>
          <li
            class="c25"
          >
            <a
              class="c26"
            >
              <div
                class="c27"
              >
                <img
                  alt=""
                  class="c28"
                  src="/static/assets/icons/time.svg"
                />
                <h3
                  class="c29"
                >
                  Prochainement
                </h3>
                <p
                  class="c30"
                >
                  Bientôt d'autres outils disponibles...
                </p>
              </div>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </main>
  <footer
    class="c32"
  >
    <div
      class="c33"
    >
      <div
        class="c1 c34"
      >
        <h2>
          Besoin d’un accompagnement personnalisé ?
        </h2>
        <div
          class="c35 c36"
        >
          Les services de renseignement en droit du travail peuvent vous donner des informations juridiques générales relatives au Code du travail, aux conventions collectives, à la jurisprudence. Ils peuvent également vous conseiller et vous orienter dans vos démarches.
        </div>
        <button
          aria-pressed="false"
          class="c37"
        >
          Contacter les services de renseignement
        </button>
      </div>
    </div>
    <div
      class="c33"
    >
      <div
        class="c1 c38"
      >
        <div
          class="c39 c40"
        >
          <h3
            class="c41"
          >
            Code du travail numérique
          </h3>
          <ul
            class="c42"
          >
            <li
              class="c43"
            >
              <a
                href="/droit-du-travail"
              >
                Le droit du travail
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="/glossaire"
              >
                Glossaire
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="/a-propos"
              >
                À propos
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="/mentions-legales"
              >
                Mentions légales
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="mailto:codedutravail@beta.gouv.fr"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
        <div
          class="c39 c40"
        >
          <h3
            class="c41"
          >
            Aidez-nous à améliorer cet outil
          </h3>
          <ul
            class="c42"
          >
            <li
              class="c43"
            >
              <a
                href="https://github.com/SocialGouv/code-du-travail-numerique/tree/vX.Y.Z"
                rel="noopener noreferrer"
                target="_blank"
              >
                Contribuer sur Github
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="https://github.com/SocialGouv/code-du-travail-numerique/releases/tag/vX.Y.Z"
                rel="noopener noreferrer"
                target="_blank"
              >
                Journal des modifications
              </a>
            </li>
          </ul>
        </div>
        <div
          class="c39 c40"
        >
          <h3
            class="c41"
          >
            En collaboration avec
          </h3>
          <ul
            class="c42"
          >
            <li
              class="c43"
            >
              <a
                href="https://travail-emploi.gouv.fr/ministere/organisation/article/dgt-direction-generale-du-travail"
                rel="noopener noreferrer"
                target="_blank"
              >
                La Direction Générale du Travail
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="https://incubateur.social.gouv.fr/"
                rel="noopener noreferrer"
                target="_blank"
              >
                L’incubateur des ministères sociaux
              </a>
            </li>
            <li
              class="c43"
            >
              <a
                href="https://beta.gouv.fr/"
                rel="noopener noreferrer"
                target="_blank"
              >
                beta.gouv.fr
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
